### DISCUSSION TITLE
Is nvim-treesitter being reimplemented in core neovim?

### DISCUSSION THREAD

**Discussion by j-xella** [nvim-treesitter/nvim-treesitter#3740] [General]
Hi,

I am new to neovim, so apologies if my question is silly. But I read on twitter about pull request [#15391](https://github.com/neovim/neovim/pull/15391) to add treesitter-based highlighting for some languages (C, VIML, LUA) in core neovim. Possibly more languages to follow. 

It got me wondering, what does it mean for this plugin then? Will those languages be dropped from this plugin, or there will be 2 treesitter implementations for those languages? Those multiple implementations, will they augment or conflict with each other?

**Comment by clason** [comment]
It means nothing for this plugin, at least in the foreseeable future. Highlighting itself is fully taken of by core Neovim. It is true that Neovim since 0.8 bundles parsers or queries for C, Lua, VimL, and Vim help, but these are not enabled by default, and nvim-treesitter still ships (possibly newer) versions of these, which will in typical situations just replace the bundled parsers and queries.
